### ISSUE TITLE
NullPointerException - Parameter specified as non-null is null (updatedKey)

### ISSUE THREAD

**Issue by abdurahmanadilovic** [russhwolf/multiplatform-settings#108]
Hi, first of all, thanks for this awesome library, we are using in our KMM project, and its working great. We do have this issue that occurs on some devices, we have seen it on android 11 and 12, and not on android 10 and 9. Here is the stack trace
```
java.lang.NullPointerException: Parameter specified as non-null is null: method com.russhwolf.settings.AndroidSettings.addListener$lambda-1, parameter updatedKey
    at com.russhwolf.settings.AndroidSettings.addListener$lambda-1
    at com.russhwolf.settings.AndroidSettings.lambda$ejYPvc8UpZPLw0Pu9AuA51-Vbq8
    at com.russhwolf.settings.-$$Lambda$AndroidSettings$ejYPvc8UpZPLw0Pu9AuA51-Vbq8.onSharedPreferenceChanged
    at android.app.SharedPreferencesImpl$EditorImpl.notifyListeners(SharedPreferencesImpl.java:629)
    at android.app.SharedPreferencesImpl$EditorImpl.lambda$notifyListeners$0$SharedPreferencesImpl$EditorImpl(SharedPreferencesImpl.java:643)
    at android.app.SharedPreferencesImpl$EditorImpl$$ExternalSyntheticLambda0.run
    at android.os.Handler.handleCallback(Handler.java:938)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loopOnce(Looper.java:226)
    at android.os.Looper.loop(Looper.java:313)
    at android.app.ActivityThread.main(ActivityThread.java:8633)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1133)
```

What's strange is that we do not attach any listeners when we are creating the Settings class in our Koin setup. So I am puzzled about how and why is Android SharedPreferences calling notifyListeners.

**Comment by russhwolf** [comment]
Huh, I guess I missed [this change](https://developer.android.com/about/versions/11/behavior-changes-11#sharedpreferences-listener) in Android 11. I'm not sure how it hasn't already come up before as this looks like it will crash anytime an app sets a listener and then later calls `clear()`. Should be a pretty easy fix, though. In the meantime, yes the workaround would be to remove listeners before calling clear.

**Comment by russhwolf** [comment]
Oh scratch that, it should be rare because `AndroidSettings.clear()` doesn't call `clear()` on the `SharedPreferences`, it calls `remove()` on every key instead. The only time you'll see this issue is if you call `preferences.edit().clear().apply()` after setting listeners on `AndroidSettings`. So a better workaround is to call `clear()` on the `Settings` instead of the `SharedPreferences`.

**Comment by russhwolf** [comment]
This is fixed in version 0.9

**Comment by abdurahmanadilovic** [comment]
Thank you for a swift response and sorry for the late reply. Keep up the good work 👍
